### ISSUE TITLE
Random appearing clock in the left upper corner

### ISSUE THREAD

**Issue by ura23** [koreader/koreader#11896]
* KOReader version: KOReader 2024.04 "Sourdough" and previous
* Device: Kindle PW5

#### Issue
During the reading randomly clock appear in the left upper corner (in Portrait mode), which disappear after page update.

#### Steps to reproduce
Just start to read book and it will appear, more likely, if you open some menus or change orientation.
 
![Reader_2023-07-22_111729](https://github.com/koreader/koreader/assets/6824577/7c380020-074a-4d0b-871e-d292a63b3af8)
![Reader_2024-05-26_133104](https://github.com/koreader/koreader/assets/6824577/aa70b88a-2c6a-4943-9109-03ea3e486594)

**Comment by Frenzie** [comment]
See #5943. Could you add some more information about the Kindle PW2 itself? (Firmware version, special offers, anything else that might be relevant.)

**Comment by ura23** [comment]
> See #5943. Could you add some more information about the Kindle PW5 itself? (Firmware version, special offers, anything else that might be relevant.)

Yes, of course. Kindle running 5.14.2 firmware, used for jailbreak WatchThis method. Device is not registered (logged in) in Amazon and probably with special offers, but I have never seen them, maybe because it is is not registered.

**Comment by AryaTwirl** [comment]
Hi, I'm also suffering from this issue. The only way so far is to start Koreader in no framework mode (although this gets mildly annoying when I want to transfer files via usb). 

I have read the #5943 issue, but not sure what I should do from there. Is there any userpatch that I can apply right away, or should I compile it first? 

Thank you.

**Comment by NiLuJe** [comment]
Nope, no framework is your best bet. I've lost any and all interest in playing whack-a-mole with random Kindle FW versions.

Patches welcome, though.

**Comment by Piecho2** [comment]
I do have the same issue on the PW12 running 5.17 and latest Koreader nightly build.

**Comment by tzzzzzzzzzzz** [comment]
This problem still exists in the latest versions of koreader and ko2

**Comment by mergen3107** [comment]
This is the same as #6659

There is nothing KOReader can do about those timings to auto refresh Kindle's UI. 

Out of scope

**Comment by sparklerfish** [comment]
I've been fiddling with this and and seems like just killing the statusbar process while KOReader is open resolves it.  Hard to 100% test/prove for sure because it's intermittent/timing related, but I added this to my koreader.sh and so far have been unable to reproduce the clock/wifi/etc bleed through since adding it.

On line 79 after the list of toggled services:
```sh
if [ -f "/etc/upstart/statusbar.conf" ]; then
    TOGGLED_SERVICES="${TOGGLED_SERVICES} statusbar"
fi
```

The bleedthrough is literally the only reason I run KOReader frameworkless, and it seems like overkill to stop and restart the GUI entirely just to prevent the statusbar leaking in, so a more lightweight solution than "just use no framework" would be great.

Let me know if there's a reason not to do it this way; otherwise if this fix makes sense I can make a PR.

**Comment by mergen3107** [comment]
Should `statusbar` be also restored upon exit from KOReader in regular mode?

**Comment by sparklerfish** [comment]
> Should `statusbar` be also restored upon exit from KOReader in regular mode?

Everything in `TOGGLED_SERVICES` gets restored already, seems to work fine when I exit.

**Comment by mergen3107** [comment]
Ah, my bad 👍🏻

**Comment by Frenzie** [comment]
Sounds fine, but include a comment in the code about the firmware version it's known to work on (cf. the whackamole comment above).

**Comment by sparklerfish** [comment]
> Sounds fine, but include a comment in the code about the firmware version it's known to work on (cf. the whackamole comment above).

That's what the check (`if [ -f "/etc/upstart/statusbar.conf" ]`) does; I don't know the exact firmwares or devices that it works on, but this checks for if it's a version that does have the KPP chrome.  I personally tested on two PW6s (5.17.1.0.4 and 5.19.4.0.1) and a PW4 (5.18.1) but I don't want to make a claim about firmware ranges that I can't back up when there's a more objective check for the actual device's config instead.

**Comment by Frenzie** [comment]
That's all I mean: "verified to work on yada yada". The file check doesn't supply such information to the reader. You definitely shouldn't include any unverified claims.

**Comment by sparklerfish** [comment]
Added comment with verified devices/versions; PR made #15846 ^_^
